CMakeLists.txt

```diff
@@ -291,6 +271,9 @@ if(BUILD_TESTING)
   if(WIN32)
     target_sources(ninja_test PRIVATE src/includes_normalize_test.cc src/msvc_helper_test.cc
       windows/ninja.manifest)
+
+    # Silence warnings about using unlink rather than _unlink
+    target_compile_definitions(ninja_test PRIVATE _CRT_NONSTDC_NO_DEPRECATE)
```

Comment:
Is there an advantage or difference here of using the generator expression versus just a normal `if(MSVC)` like it was before?
My understanding is since we know during the configuration step if MSVC is defined, we can just use a normal if and the generator provides nothing extra - please inform me if this isn't the case